src/NumericalAlgorithms/Spectral/SwshTransformJob.hpp

```diff
@@ -0,0 +1,406 @@
+ * carefully. (In particular, it does not use the sign convention you will find
+ * in [Wikipedia]
+ * (https://en.wikipedia.org/wiki/Spin-weighted_spherical_harmonics)).
+ * - libsharp deals only with the transformation of real values, so
+ *   transformation of complex values must be done in multiple components.
+ * - due to only transforming real components, it stores only the positive
+ *   \f$m\f$ modes (as the rest would be redundant). Therefore, the negative
+ *   \f$m\f$ modes must be inferred from conjugation and further sign changes.
+ * - libsharp only has the capability of transforming positive spin weighted
+ *   quantities. Therefore, additional steps are taken (involving further
+ *   conjugation of the provided data) in order to use those utilities on
+ *   negative spin-weighted quantities. The resulting modes have yet more sign
+ *   changes that must be taken into account.
+ *
+ * The decomposition resulting from the libsharp transform for spin \f$s\f$ and
+ * complex spin-weighted \f${}_s f\f$ can be represented mathematically as:
+ *
+ * \f{align*}{
+ * {}_s f(\theta, \phi) = \sum_{\ell = 0}^{l_\mathrm{max}} \Bigg\{&
+ * \left(\sum_{m = 0}^{\ell} a^\mathrm{sharp, real}_{l m} {}_s Y_{\ell
+ * m}^\mathrm{sharp, real}\right) + \left(\sum_{m=1}^{\ell}
+ * \left(a^\mathrm{sharp, real}_{l m}{}\right)^*
+ * {}_s Y_{\ell -m}^\mathrm{sharp, real}\right) \notag\\
+ * &+ i \left[\left(\sum_{m = 0}^{\ell} a^\mathrm{sharp, imag}_{l m} {}_s
+ * Y_{\ell m}^\mathrm{sharp, imag}\right) + \left(\sum_{m=1}^{\ell}
+ * \left(a^\mathrm{sharp, imag}_{l m}{}\right)^* {}_s Y_{\ell -m}^\mathrm{sharp,
+ * imag} \right)\right] \Bigg\},
+ * \f}
+ *
+ * where
+ *
+ * \f{align*}{
+ * {}_s Y_{\ell m}^\mathrm{sharp, real} &=
+ * \begin{cases}
+ * (-1)^{s + 1} {}_s Y_{\ell m}, & \mathrm{for}\; s < 0, m >=0 \\
+ * {}_s Y_{\ell m}, & \mathrm{for}\; s = 0, m >=0 \\
+ * - {}_s Y_{\ell m}, & \mathrm{for}\; s > 0, m >=0 \\
+ * (-1)^{s + m + 1} {}_s Y_{\ell m}, & \mathrm{for}\; s < 0, m < 0 \\
+ * (-1)^{m} {}_s Y_{\ell m}, & \mathrm{for}\; s = 0, m < 0 \\
+ * (-1)^{m + 1} {}_s Y_{\ell m}, & \mathrm{for}\; s > 0, m < 0
+ * \end{cases} \\
+ * {}_s Y_{\ell m}^\mathrm{sharp, imag} &=
+ * \begin{cases}
+ * (-1)^{s + 1} {}_s Y_{\ell m}, & \mathrm{for}\; s < 0, m >=0 \\
+ * -{}_s Y_{\ell m}, & \mathrm{for}\; s = 0, m >=0 \\
+ * {}_s Y_{\ell m}, & \mathrm{for}\; s > 0, m >=0 \\
+ * (-1)^{s + m} {}_s Y_{\ell m}, & \mathrm{for}\; s < 0, m < 0 \\
+ * (-1)^{m} {}_s Y_{\ell m}, & \mathrm{for}\; s = 0, m < 0 \\
+ * (-1)^{m + 1} {}_s Y_{\ell m}, & \mathrm{for}\; s > 0, m < 0
+ * \end{cases},
+ * \f}
+ *
+ * where the unadorned \f${}_s Y_{lm}\f$ on the right-hand-sides follow the
+ * (older) convention of \cite Goldberg1966uu. Note the phase in these
+ * expressions is not completely standardized, so should be checked carefully
+ * whenever coefficient data is directly manipulated.
+ *
+ * The resulting coefficients are stored in a triangular,
```

Comment:
"coefficients" -> "coefficients \f$a_{\ell m}\f$"  (assuming that's what the transform returns).